stablehlo/tests/ops_stablehlo.mlir

```diff
@@ -945,128 +945,117 @@ func.func @broadcast_in_dim_unranked_operand(%arg0 : tensor<*xf32>) -> tensor<2x
 
 // -----
 
-// CHECK-LABEL: if
-func.func @if(%pred : tensor<i1>, %branch_operand : tensor<2xf32>) {
-  %0 = "stablehlo.if"(%pred) ({
-      "stablehlo.return"(%branch_operand) : (tensor<2xf32>) -> ()
-    }, {
-      "stablehlo.return"(%branch_operand) : (tensor<2xf32>) -> ()
-    }) : (tensor<i1>) -> tensor<2xf32>
-  func.return
+// CHECK-LABEL: func @if
+func.func @if(%pred : tensor<i1>, %branch_operand : tensor<f32>) -> (tensor<f32>, tensor<f32>) {
+  %0, %1 = "stablehlo.if"(%pred) ({
+      "stablehlo.return"(%branch_operand, %branch_operand) : (tensor<f32>, tensor<f32>) -> ()
+  }, {
+      %2 = "stablehlo.if"(%pred) ({
+          "stablehlo.return"(%pred) : (tensor<i1>) -> ()
+      }, {
+          "stablehlo.return"(%pred) : (tensor<i1>) -> ()
+      }) : (tensor<i1>) -> tensor<i1>
+      "stablehlo.return"(%branch_operand, %branch_operand) : (tensor<f32>, tensor<f32>) -> ()
+  }) : (tensor<i1>) -> (tensor<f32>, tensor<f32>)
+  func.return %0, %1 : tensor<f32>, tensor<f32>
```

Comment:
Why do we need a more complicated test here? Does it test something that the simpler test didn't test?

Reply:
There were two positive tests, one with and another without nested `if` operations. I decided to use the more complex one to remove "duplicate" positive tests. If the purpose of the positive tests in this file is for coherence check that the op works, then I think it's fine to include a simpler one. Another option is to have both tests, and I don't have preference for either option. What are your thoughts?

Reply:
I think that minimizing the number and the composition of tests is important for maintainability. We have only captured this in https://github.com/openxla/stablehlo/blob/main/docs/reference.md#testing-guidelines, but I think this also applies to ops_stablehlo.mlir.